src/modules/transactions/hooks/create/useCreateTransaction.ts

```diff
@@ -267,6 +273,58 @@ const useCreateTransaction = (props?: UseCreateTransactionParams) => {
     [],
   );
 
+  const transactionsForm = useWatch({
+    control: form.control,
+    name: 'transactions',
+    defaultValue: [],
+  });
+
+  const allAssetsUsed = useMemo(() => {
+    const nfts =
+      props?.nfts?.map((nft) => ({
+        assetId: nft.assetId,
+      })) ?? [];
+
+    const assets =
+      currentVaultAssets?.map((asset) => ({
+        assetId: asset.assetId,
+        units: asset.units,
+      })) ?? [];
+
+    const allNftsInTransactions =
+      nfts.length === 0 ||
+      nfts.every((nft) =>
+        (transactionsForm ?? []).some(
+          (transaction) => transaction.asset === nft.assetId,
+        ),
+      );
+
+    const hasSufficientBalance = assets.every(({ assetId, units }) => {
+      const available = getBalanceAvailable(assetId);
+      const assetAmount = bn.parseUnits(available, units);
```

Comment:
```suggestion
    // Early return if no transactions
    if (!transactionsForm?.length) return false;
    
    // Extract transaction assets for more efficient lookups
    const transactionAssetIds = new Set(
      transactionsForm.map(transaction => transaction.asset)
    );

    // Check if all NFTs are included in transactions
    const nfts = props?.nfts || [];
    const allNftsInTransactions = nfts.length === 0 || 
      nfts.every(nft => transactionAssetIds.has(nft.assetId));
      
    const assets = currentVaultAssets || [];
    
    const hasSufficientBalance = assets.every(({ assetId, units }) => {
      // If asset not in transactions, return false (not all assets used)
      if (!transactionAssetIds.has(assetId)) return false;
```
I think it's work more efficient because avoid array manipulations